### ISSUE TITLE
total loss 很大 ，并且下降的特别慢，跪求各位大佬指点

### ISSUE THREAD

**Issue by leiqing110** [argusswift/YOLOv4-pytorch#147]
![image](https://user-images.githubusercontent.com/19700402/107218334-7223b900-6a4a-11eb-988c-ac0c80437326.png)

**Comment by jingtianyilong** [comment]
Your settings? And your data?

**Comment by leiqing110** [comment]
> Your settings? And your data?

Thank you for your reply, I'm using the customer dataset, this is my configuration
![Snipaste_2021-02-09_17-29-04](https://user-images.githubusercontent.com/19700402/107343673-87592000-6afc-11eb-95a8-8a6883c72cd4.png)

**Comment by jingtianyilong** [comment]
Batch size 2 is too small imo. Network hardly learn anything. I normally use 16. 8 is also a good size. But this relate largely to your GPU. If 2 is what you can do best. Then we stick to it and try other params.
10 epochs is way smaller than it should be. Start with about 100 and start with larger `lr_init`. If too large, shrink it down a little. Change `warmup_epochs` in case you see strange loss curve at the beginning.

**Comment by leiqing110** [comment]
> Batch size 2 is too small imo. Network hardly learn anything. I normally use 16. 8 is also a good size. But this relates largely to your GPU. If 2 is what you can do best. Then we stick to it and play with other params.
> 10 epochs is way smaller than it should be. Start with about 100 and start with larger `lr_init`. If too large, shrink it down a little. Change `warmup_epochs` in case you see strange loss curve at the beginning.
> You can switch off `multi_scale_train` during you trail. This accelarate your testing process a little.

Thank you again, I'll retrain your method

**Comment by leiqing110** [comment]
After retraining, there was an error in the validation process, and I haven't solved the problem yet, you know?
![Snipaste_2021-02-15_09-56-56](https://user-images.githubusercontent.com/19700402/107897837-28206300-6f75-11eb-983e-c9d0e2fc7a24.png)

**Comment by leiqing110** [comment]
> Batch size 2 is too small imo. Network hardly learn anything. I normally use 16. 8 is also a good size. But this relates largely to your GPU. If 2 is what you can do best. Then we stick to it and play with other params.
> 10 epochs is way smaller than it should be. Start with about 100 and start with larger `lr_init`. If too large, shrink it down a little. Change `warmup_epochs` in case you see strange loss curve at the beginning.
> You can switch off `multi_scale_train` during you trail. This accelarate your testing process a little.

After retraining, there was an error in the validation process, and I haven't solved the problem yet, you know?
![Snipaste_2021-02-15_09-56-56](https://user-images.githubusercontent.com/19700402/107897905-556d1100-6f75-11eb-9b28-edf004cbc73c.png)

**Comment by jingtianyilong** [comment]
I think the error is quiet self-explanatory. 
TypeError

**Comment by leiqing110** [comment]
> I think the error is quiet self-explanatory.
> TypeError

![image](https://user-images.githubusercontent.com/19700402/108442191-945ed780-7291-11eb-9c4e-e62d8f774f73.png)
训练了100epoch ，最低的loss为105左右 。执行evel_voc.py 之后，ap都为0，心态崩了。。。
![image](https://user-images.githubusercontent.com/19700402/108442332-d4be5580-7291-11eb-8477-44a14b2bccfe.png)

**Comment by jingtianyilong** [comment]
It didn't learn anything. Loss should be in 1 digit in most of the cases. At least according to my experience.
Try different batch size and learning rate. Those matters a lot. **_Track your loss curve. Track your eval during your training._**
You don't have to train all the way to see if the training is broken. For me 20 epochs should be enough for you to see if it won't learn anything.

**Comment by chennaikai11** [comment]
请问解决了吗 我也是这个问题 一样的数据集

**Comment by abhigoku10** [comment]
@jingtianyilong @leiqing110  has this issue solved ?
